### ISSUE TITLE
SDR assignment operator= not working as expected

### ISSUE THREAD

**Issue by breznak** [htm-community/nupic.cpp#425]
```
SDR copy = (SDR)orig;
```
copy does inherit data, but not dimensions. 
It would be expected that after `op=` both SDRs are equal. 

Discovered in 088d4a7

**Comment by ctrl-z-9000-times** [comment]
The SDR currently does not implement the assignment `operator=` so i dont know how this compiles.  Implementing it should fix this bug.

**Comment by breznak** [comment]
> The SDR currently does not implement the assignment operator=

It does: 

```
SparseDistributedRepresentation& operator=(const SparseDistributedRepresentation& value)
{ setSDR(value); return *this; }
```

Might be related to the problems with SDR being cast-to from a vector (?)

**Comment by ctrl-z-9000-times** [comment]
Thanks for finding this bug. I will change it to initialize the SDR if it isnt already, and check dimensions if it is initialized. Im away from my workstation for a while so i will get to this in a few days.  I will include the testcase youve written too, thanks for making a reproducer.

**Comment by ctrl-z-9000-times** [comment]
Fixed in PR #426.
